core/cthreads.h

```diff
@@ -0,0 +1,91 @@
+#ifndef CTHREADS_H
+#define CTHREADS_H
+
+struct cthreads_args {
+  void *(*func)(void *data);
+  void *data;
+};
+
+#if __unix__
+#include <pthread.h>
+#elif _WIN32
+#include <windows.h>
+#endif
+
+struct cthreads_thread {
+    #ifdef __unix__
+    pthread_t pThread;
+    #elif _WIN32
+    HANDLE wThread;
+    #endif
+};
```

Comment:
What if you did this instead?
```suggestion
#ifdef __unix__
    typedef pthread_t cthreads_thread_t;
#elif _WIN32
    typedef HANDLE cthreads_thread_t;
#endif
```
Then there's no need to access the struct field.. if you think this is a good idea, you should also do it for mutex, cond, etc

Reply:
Only if you wish to, for me both are awesome